posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+
+image::/img/blog/jenkins-cred-binding-and-corresponding-param.png[Pipeline credential binding and corresponding param,width=30%,align="center"]
+
+* code identifier: This is optional, a unique docker image identifier
+* docker image version: docker image version number, a unique identifier   
+
+*Security scan* can be done along with Maven build command (CLI commands way of application scanning) or can do it in a separate pipeline stage. 
+
+For static code analysis, SonarQube community edition can be used. Install SonarQube server by either using file start-up type from cli by downloading the package in link:https://www.sonarqube.org/success-download-community-edition/[SonarQube server installation package] or use docker way as explained in link:https://docs.sonarqube.org/latest/setup/get-started-2-minutes/[SonarQube server installation steps]. SonarQube jenkins client setup details are given at link:https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-jenkins/[SonarQube client for jenkins]. 
+ 
+    CLI command example, integrated with Maven command: 
+    $ mvn package verify sonar:sonar -Dsonar.projectKey=sampleapp -Dsonar.host.url=http://localhost:9000 -Dsonar.login=7b13f240123456780479b13d09ba8
+   
+For Docker image scan, you can use link:https://hub.docker.com/r/aquasec/trivy/[Docker image scan with trivy]. This scan provides the vulnerability details of open source jars that are used in the application. Sample commands are given below
+
+       sh '''
+             docker run aquasec/trivy image $docker_login/sampleapp:v1.0
+             #eg: docker run aquasec/trivy image docker.io/someid/liberty-app:v1.0
+       '''
+
+
+== Deploying Jakarta EE & MicroProfile applications with Open Liberty to Kubernetes with Jenkins
+
+For simplicity, I will use the command line (CLI) option to configure Jenkins to deploy a Jakarta EE & MicroProfile application with Open Liberty to Kubernetes. You could use one of many tools though, such as Helm, Travis CI, Circle CI, etc.
+
+Create a new stage in the pipeline code and write all the needed commands between the shell option (sample commands give below):
+                           
+                           sh '''
+                              ibmcloud login --apikey $IBM_CLOUD_API_KEY -g $IBM_CLOUD_RSGRP
+                              ibmcloud ks cluster config --cluster $CLUSTER-ID
+                              kubectl config current-context
+                              kubectl create -f deploy/deployment.yaml #( simple k8s deployment command )
+                              kubectl create -f deploy/service.yaml #( simple k8s deployment command )
+                              kubectl create -f deploy/route.yaml #( simple k8s deployment command )
+                           '''
+
+Make sure that your Kubernetes configuration files are in the same source code repository (under a folder called "deploy"), also the image name should be updated manually or programmatically 
```

Comment:
Change to "Make sure that your Kubernetes configuration files are stored in the same git repository as your Jenkinsfile in a sub-directory called `deploy`." (end of sentence)

I don't understand the second sentence (from "Also the image name..."). Who should update the image name and why and when?

Reply:
Modified to "Make sure that your Kubernetes configuration files are stored in the same git repository as your Jenkinsfile in a sub-directory called deploy, also the docker image name in kubernetes deployment configuration file (as shown in screenshot below) should be updated manually or programmatically (in case of any change in it on run time)Make sure that your Kubernetes configuration files are stored in the same git repository as your Jenkinsfile in a sub-directory called deploy, also the docker image name in kubernetes deployment configuration file (as shown in screenshot below) should be updated manually or programmatically (in case of any change in it on run time)

Reply:
I can explain further when you are back